File: go/258_Add_Digits/258_Add_Digits.go
func addDigits(num int) int {
    for num > 9 {
        num = sumDigits(num)
    }
    return num
}

func sumDigits(n int) int {
    total := 0
    for n > 0 {
        digit := n % 10
        n = n / 10
        total += digit
    }
    return total
}

File: java/258_Add_Digits/258_Add_Digits.java
class Solution {
    public int addDigits(int num) {
        while (num > 9) {
            num = sumDigits(num);
        }
        return num;
    }
    
    public int sumDigits(int n) {
        int total = 0;
        while (n > 0) {
            total += n % 10;
            int digit = n % 10;
            n = n / 10;
        }
        return total;
    }
}
